trl/trainer/utils.py

```diff
@@ -743,13 +743,23 @@ def get_global_statistics(
 
 def compute_accuracy(eval_pred) -> Dict[str, float]:
     predictions, labels = eval_pred
-    # Here, predictions is rewards_chosen and rewards_rejected.
-    # We want to see how much of the time rewards_chosen > rewards_rejected.
-    if np.array(predictions[:, 0] == predictions[:, 1], dtype=float).sum() > 0:
-        warnings.warn(
-            f"There are {np.array(predictions[:, 0] == predictions[:, 1]).sum()} out of {len(predictions[:, 0])} instances where the predictions for both options are equal. As a consequence the accuracy can be misleading."
+    if predictions.ndim == 3:
```

Comment:
We need to have a test for this new case

Reply:
Yes, there is already one here: https://github.com/gaetanlop/trl/blob/c3eb08ea31951a40f1d7cad45e4ff433bc5fa5c2/tests/test_stepwise_reward_trainer.py#L39-L45

Shouldn't it be enough given that the `RewardTrainer` also has a test when ndim==2?